idb/dummy.go

```diff
@@ -516,19 +543,103 @@ type RoundUpdates struct {
 
 // Clear is used to set a RoundUpdates object back to it's default values.
 func (ru *RoundUpdates) Clear() {
-	ru.AlgoUpdates = nil
-	ru.AccountTypes = nil
-	ru.AccountDataUpdates = nil
+	//ru.AlgoUpdates = nil
+	//ru.AccountTypes = nil
+	//ru.AccountDataUpdates = nil
+	//ru.AssetUpdates = nil
+
+	ru.AlgoUpdates = make(map[[32]byte]*AlgoUpdate)
+	ru.AccountTypes = make(map[[32]byte]string)
+	ru.AccountDataUpdates = make(map[[32]byte]map[string]interface{})
 	ru.AcfgUpdates = nil
 	ru.TxnAssetUpdates = nil
-	ru.AssetUpdates = nil
+	ru.AssetUpdates = make(map[[32]byte][]AssetUpdate)
 	ru.FreezeUpdates = nil
 	ru.AssetCloses = nil
 	ru.AssetDestroys = nil
 	ru.AppGlobalDeltas = nil
 	ru.AppLocalDeltas = nil
 }
 
+type UpdateFilter struct {
+	StartRound int64
+	RoundLimit *int
+	Address    *types.Address
+}
+
+// Filter is used to get a RoundUpdates copy containing only particular RoundUpdates.
+func (ru *RoundUpdates) Filter(filter UpdateFilter) RoundUpdates {
```

Comment:
I think this is the most likely spot where I have bugs. I need to ensure that only `account`, `account_app`, and `account_asset` are modified.

While writing this comment I realized that the `CommitRoundAccounting` probably needs to be changed, it needs to avoid setting the accounting round when re-indexing.